vochain/process.go

```diff
@@ -484,56 +527,69 @@ func (app *BaseApplication) NewProcessTxCheck(vtx *models.Tx, txBytes,
 }
 
 // SetProcessTxCheck is an abstraction of ABCI checkTx for canceling an existing process
-func SetProcessTxCheck(vtx *models.Tx, txBytes, signature []byte, state *State) error {
+func SetProcessTxCheck(vtx *models.Tx, txBytes, signature []byte, state *State) (common.Address, error) {
 	tx := vtx.GetSetProcess()
 	// check signature available
 	if signature == nil || tx == nil || txBytes == nil {
-		return fmt.Errorf("missing signature on setProcess transaction")
+		return common.Address{}, fmt.Errorf("missing signature on setProcess transaction")
 	}
-
-	// Check if transaction owner has enough funds to create a process
-	authorized, addr, err := state.VerifyAccountBalance(txBytes, signature, SetProcessCost)
+	// Check if transaction owner has enough funds to set a process
+	cost, err := state.TxCost(tx.Txtype, false)
 	if err != nil {
-		return err
+		return common.Address{}, fmt.Errorf("cannot get tx cost from state %w", err)
+	}
+	sigAddress, acc, err := state.AccountFromSignature(txBytes, signature)
+	if err != nil {
+		return common.Address{}, fmt.Errorf("cannot get account from signature %w", err)
 	}
 	// get process
 	process, err := state.Process(tx.ProcessId, false)
 	if err != nil {
-		return fmt.Errorf("cannot get process %x: %w", tx.ProcessId, err)
+		return common.Address{}, fmt.Errorf("cannot get process %x: %w", tx.ProcessId, err)
 	}
+
 	isOracle := false
+	authorized := acc.Balance >= cost
 	if authorized {
-		// If owner authorized, check entityId matches with owner address
-		if !bytes.Equal(process.EntityId, addr.Bytes()) {
-			return fmt.Errorf("process entityID and transaction owner do not match")
-		}
-	} else {
-		// Check if the transaction comes from an oracle
-		// Oracles can create processes with any entityID
-		if isOracle, err = state.IsOracle(addr); err != nil {
-			return err
+		// If address authorized, check entityId matches with address address or a delegate
```

Comment:
this logic looks very similar to line 455 (`// check tx signer can create process`)
maybe make a function out of them?

Reply:
i see you need isOracle further down, so i don't know. maybe it's not worth it.